services/CurrentTime/source/CurrentTimeService.cpp

```diff
@@ -42,19 +42,55 @@ ble_error_t CurrentTimeService::init()
     return bleError;
 }
 
-time_t CurrentTimeService::get_time() {
+void CurrentTimeService::set_event_handler(EventHandler *handler) {
+    _current_time_handler = handler;
+}
+
+time_t CurrentTimeService::get_time()
+{
     time_t epoch_time = time(nullptr);
 
     return epoch_time + _time_offset;
 }
 
-void CurrentTimeService::set_time(time_t new_time) {
+void CurrentTimeService::set_time(time_t new_time)
+{
     time_t epoch_time = time(nullptr);
 
     _time_offset = new_time - epoch_time;
 }
 
-void CurrentTimeService::onDataRead(GattReadAuthCallbackParams *read_request) {
+bool CurrentTimeService::current_time_is_valid()
+{
+    bool result = true;
+
+    if ((_current_time.year    < 1582) || (_current_time.year    > 9999)) {
+        result = false;
```

Comment:
don't continue checking all the other if conditions if you already know it's wrong, just return false here and now